Error trying to install on linux

```r
install.packages("processx")
```
ends with error message
```
unix/processx.c: In function 'processx__make_socketpair':
unix/processx.c:232:41: error: 'SOCK_CLOEXEC' undeclared (first use in this function)
   if (socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, pipe) == 0)
                                         ^
unix/processx.c:232:41: note: each undeclared identifier is reported only once for each function it appears in
/share/data/apps/R/R-3.5.0/etc/Makeconf:159: recipe for target 'unix/processx.o' failed
make: *** [unix/processx.o] Error 1
ERROR: compilation failed for package ‘processx’
* removing ‘/share/data/apps/R/R-3.5.0/library/processx’
```
```
sessioninfo::platform_info()
 setting  value
 version  R version 3.5.0 (2018-04-23)
 os       <NA>
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 tz       America/Vancouver
 date     2018-05-15
```
```
$cat /etc/*-release
CentOS release 5.11 (Final)
```

Maybe related, I am getting the following error:

```
trying URL 'https://cloud.r-project.org/src/contrib/processx_3.1.0.tar.gz'
Content type 'application/x-gzip' length 78448 bytes (76 KB)
==================================================
downloaded 76 KB

* installing *source* package ‘processx’ ...
** package ‘processx’ successfully unpacked and MD5 sums checked
** libs
gcc -O2 -march=native -std=c99  -Wall tools/px.c -o tools/px
tools/px.c: In function ‘main’:
tools/px.c:76:21: warning: implicit declaration of function ‘usleep’ [-Wimplicit-function-declaration]
       if (fnum > 0) usleep((useconds_t) (fnum * 1000.0 * 1000.0));
                     ^~~~~~
tools/px.c:76:29: error: ‘useconds_t’ undeclared (first use in this function)
       if (fnum > 0) usleep((useconds_t) (fnum * 1000.0 * 1000.0));
                             ^~~~~~~~~~
tools/px.c:76:29: note: each undeclared identifier is reported only once for each function it appears in
make: *** [tools/px] Error 1
ERROR: compilation failed for package ‘processx
```

I am working on a machine running CentOS release 7.4.1708

```
sessionInfo()
R version 3.4.1 (2017-06-30)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)
```

Is there something I can do?
I already tried adding the lines you mentioned, but did not see any change.

--- a/src/unix/processx-unix.h
+++ b/src/unix/processx-unix.h
@@ -2,6 +2,14 @@
 #ifndef R_PROCESSX_UNIX_H
 #define R_PROCESSX_UNIX_H
 
+# ifndef O_CLOEXEC
+#  define O_CLOEXEC 02000000
+# endif
+
+# ifndef SOCK_CLOEXEC
+#  define SOCK_CLOEXEC O_CLOEXEC
+# endif
+
 typedef struct processx_handle_s {
   int exitcode;
   int collected;	 /* Whether exit code was collected already */

